lib/galaxy/datatypes/text.py

```diff
@@ -911,3 +912,115 @@ def _looks_like_yaml(self, file_prefix: FilePrefix):
+            "+-------------------------------------------------+",
+            "|                                                 |",
+            "|      CCC   AA    SSS  TTTTT  EEEEE  PPPP        |",
+            "|     C     A  A  S       T    E      P   P       |",
+            "|     C     AAAA   SS     T    EEE    PPPP        |",
+            "|     C     A  A     S    T    E      P           |",
+            "|      CCC  A  A  SSS     T    EEEEE  P           |",
+            "|                                                 |",
+            "+-------------------------------------------------+"
+        ]
+        handle = file_prefix.string_io()
+        for header_line in castep_header:
+            if handle.readline().strip() != header_line:
+                return False
+        return True
+
+
+@build_sniff_from_prefix
+class Param(Yaml):
+    """General input file (text)"""
+
+    file_ext = "param"
+    sniff: typing.Callable
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """
+        Modified version of the normal Yaml sniff that also checks the
+        dict keys for "general" CASTEP keywords, which are not case
+        sensitive
+        """
+        general_keywords = [
+            "BACKUP_INTERVAL",
+            "CALCULATE_STRESS",
+            "CALCULATE_DENSDIFF",
+            "CALCULATE_ELF",
+            "CALCULATE_HIRSHFELD",
+            "CHARGE_UNIT",
+            "CHECKPOINT",
+            "COMMENT",
+            "CONTINUATION",
+            "DATA_DISTRIBUTION",
+            "IPRINT",
+            "NUM_BACKUP_ITER",
+            "OPT_STRATEGY",
+            "PAGE_WVFNS",
+            "PRINT_CLOCK",
+            "PRINT_MEMORY_USAGE",
+            "RAND_SEED",
+            "REUSE",
+            "RUN_TIME",
+            "STOP",
+            "TASK",
+            "WRITE_CHECKPOINT",
+            "WRITE_FORMATTED_DENSITY",
+            "WRITE_FORMATTED_ELF",
+            "WRITE_FORMATTED_POTENTIAL",
+            "WRITE_ORBITALS"
+        ]
+        # Pattern used by SequenceSplitLocations
+        if file_prefix.file_size < 50000 and not file_prefix.truncated:
+            # If the file is small enough - don't guess just check.
+            try:
+                item = yaml.safe_load(file_prefix.contents_header)
+                assert isinstance(item, dict)
+            except yaml.YAMLError:
+                return False
+        else:
+            # If file is too big, load the first part. Trim the current line, in case it cut off in the middle of a key.
+            file_start = file_prefix.string_io().read(50000).strip().rsplit("\n", 1)[0]
+            try:
+                item = yaml.safe_load(file_start)
+                assert isinstance(item, dict)
+            except yaml.YAMLError:
+                return False
+
+        for key in item.keys():
+            assert isinstance(key, str)
+            if key.upper() in general_keywords:
+                return True
+        return False
```

Comment:
I can imagine us getting false positives from keys like "COMMENT" and "TASK". But we can't just "invert" this to reject anything that isn't in `general_keywords`, since there are a lot more possible keywords beyond the ones listed. 

Perhaps a better strategy is to check that one of the [possible values for the TASK keyword](https://www.tcm.phy.cam.ac.uk/castep/documentation/WebHelp/content/modules/castep/keywords/k_task_castep.htm) is present – I'm semi-sure TASK is the only mandatory keyword (the calculation is kind of meaningless otherwise, but I'll check this), and those values are a lot more distinctive.